library/core/src/result.rs

```diff
@@ -1214,6 +1214,41 @@ impl<T, E> Result<Result<T, E>, E> {
     }
 }
 
+impl<T> Result<T, T> {
+    /// Returns the [`Ok`] value if `self` is `Ok`, and the [`Err`] value if
+    /// `self` is `Err`.
+    ///
+    /// In other words, this function returns the value (the `T`) of a
+    /// `Result<T, T>`, regardless of whether or not that result is `Ok` or
+    /// `Err`.
+    ///
+    /// This can be useful in conjunction with APIs such as
+    /// [`Atomic*::compare_exchange`], or [`slice::binary_search`][binary_search], but only in
+    /// cases where you don't care if the result was `Ok` or not.
+    ///
+    /// [`Atomic*::compare_exchange`]: crate::sync::atomic::AtomicBool::compare_exchange
+    /// [binary_search]: ../../std/primitive.slice.html#method.binary_search
```

Comment:
Note that:

- I can't use the fancy intradoc links stuff here for reasons that only @jyn514 comprehends.
- Apparently primitives are only documented in std (#80334 or something), so I have to link it in `std`.

That said, I'm also okay with getting rid of these links entirely if that's desirable.

Reply:
You can now use intra-doc links here!

```suggestion
    /// [binary_search]: slice::binary_search
```

Reply:
Actually, better yet you can change the part above that has the link to just ``[`slice::binary_search`]`` and remove this line.

Reply:
> You can now use intra-doc links here!

No, the bootstrap bump hasn't happened yet. Intra-doc pointers landed in https://github.com/rust-lang/rust/pull/80181, which is in 1.51, which is still in nightly.

Reply:
Dang, I always forget about the bootstrap bump :(